### PR TITLE
enhancement(integration): posthog new properties added

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

list of the new properties that is added to postog : 

- $viewport_height
- $viewport_width
- utm_source
- utm_medium
- utm_campaign
- utm_content
- utm_term
- $performance_page_loaded

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [x] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/posthog/data/PHPropertiesConfig.json [hunk 1]
```diff
@@ -37,30 +37,20 @@
   },
   {
     "destKey": "$time",
-    "sourceKeys": [
-      "originalTimestamp",
-      "timestamp"
-    ]
+    "sourceKeys": ["originalTimestamp", "timestamp"]
```
Comment (shrouti1507) [inline v0/destinations/posthog/data/PHPropertiesConfig.json:40]: we can add it from generic field mapping. Can you check for the other implementations and see how to do it.

File: v0/destinations/posthog/data/PHPropertiesConfig.json [hunk 2]
```diff
@@ -140,5 +127,41 @@
     "destKey": "$app_build",
     "sourceKeys": "context.app.build",
     "isMobile": true
+  },
+  {
+    "destKey": "$viewport_height",
+    "sourceKeys": "context.viewport.height"
```
Comment (shrouti1507) [inline v0/destinations/posthog/data/PHPropertiesConfig.json:133]: There is no field called `context.viewport`. The height of the user's actual browser window (in pixels). We can map it from `traits`, or from `context.page`. WDYT @utsabc?

File: v0/destinations/posthog/data/PHPropertiesConfig.json [hunk 3]
```diff
@@ -140,5 +127,41 @@
...
+  },
+  {
+    "destKey": "$viewport_width",
```
Comment (shrouti1507) [inline v0/destinations/posthog/data/PHPropertiesConfig.json:136]: same as the above.

File: v0/destinations/posthog/data/PHPropertiesConfig.json [hunk 4]
```diff
@@ -140,5 +127,41 @@
...
+    "destKey": "utm_source",
+    "sourceKeys": "context.campaign.source"
+  },
+  {
+    "destKey": "utm_medium",
+    "sourceKeys": "context.campaign.medium"
+  },
+  {
+    "destKey": "utm_campaign",
+    "sourceKeys": "context.campaign.campaign"
```
Comment (shrouti1507) [inline v0/destinations/posthog/data/PHPropertiesConfig.json:149]: make it `context.campaign.name`

File: v0/destinations/posthog/data/PHPropertiesConfig.json [hunk 5]
```diff
@@ -140,5 +127,41 @@
...
+  {
+    "destKey": "utm_content",
+    "sourceKeys": "context.campaign.content"
+  },
+  {
+    "destKey": "utm_term",
+    "sourceKeys": "context.campaign.term"
+  },
+  {
+    "destKey": "utm_name",
```
Comment (shrouti1507) [inline v0/destinations/posthog/data/PHPropertiesConfig.json:160]: @mihir-4116 can you please refer to the doc where this property name is mentioned? 
I am referring to:
1. https://github.com/PostHog/posthog/blob/0c65195a9d1f23690a64457130f4ca1709ded4fc/frontend/src/lib/components/PropertyKeyInfo.tsx#L327
2. https://posthog.com/manual/utm-segmentation

I do not see it being mentioned over there. Am I missing something?

File: v0/destinations/posthog/data/PHPropertiesConfig.json (hunk 1 above)
Comment (mihir-4116) [reply v0/destinations/posthog/data/PHPropertiesConfig.json:40]: i don't get this. could you please explain bit more @shrouti1507 .

File: v0/destinations/posthog/data/PHPropertiesConfig.json (hunk 2 above)
Comment (mihir-4116) [reply v0/destinations/posthog/data/PHPropertiesConfig.json:133]: updated mapping. mapped it with traits

File: v0/destinations/posthog/data/PHPropertiesConfig.json (hunk 5 above)
Comment (mihir-4116) [reply v0/destinations/posthog/data/PHPropertiesConfig.json:160]: @shrouti1507 i was able to see this property in doc, but it's not showing anymore. i am removing it and update the doc card accordingly.

File: v0/destinations/posthog/data/PHPropertiesConfig.json [hunk 6]
```diff
@@ -140,5 +128,37 @@
     "destKey": "$app_build",
     "sourceKeys": "context.app.build",
     "isMobile": true
+  },
+  {
+    "destKey": "$viewport_height",
+    "sourceKeys": ["traits.viewport.height", "context.traits.viewport.height"]
```
Comment (utsabc) [inline v0/destinations/posthog/data/PHPropertiesConfig.json:134]:
```suggestion
    "sourceKeys": ["context.screen.height"]
```

File: v0/destinations/posthog/data/PHPropertiesConfig.json [hunk 7]
```diff
@@ -140,5 +128,37 @@
...
+  },
+  {
+    "destKey": "$viewport_width",
+    "sourceKeys": ["traits.viewport.width", "context.traits.viewport.width"]
```
Comment (utsabc) [inline v0/destinations/posthog/data/PHPropertiesConfig.json:138]:
```suggestion
    "sourceKeys": ["context.screen.width"]
```

File: v0/destinations/posthog/data/PHPropertiesConfig.json [hunk 8]
```diff
@@ -102,7 +90,7 @@
     "isMobile": true
   },
   {
-    "destKey": "$user_agent",
+    "destKey": "$useragent",
```
Comment (utsabc) [inline v0/destinations/posthog/data/PHPropertiesConfig.json:93]: Is this correct? Are we seeing correct values in destination?

File: v0/destinations/posthog/data/PHPropertiesConfig.json (hunk 7 above)
Comment (mihir-4116) [reply v0/destinations/posthog/data/PHPropertiesConfig.json:138]: @utsabc we already have these sourceKeys for $screen_width and $screen_height properties

File: v0/destinations/posthog/data/PHPropertiesConfig.json (hunk 8 above)
Comment (mihir-4116) [reply v0/destinations/posthog/data/PHPropertiesConfig.json:93]: Is this correct?  -> yes
Are we seeing correct values in destination? -> yes

File: v0/destinations/posthog/data/PHPropertiesConfig.json (hunk 6 above)
Comment (mihir-4116) [reply v0/destinations/posthog/data/PHPropertiesConfig.json:134]: @utsabc we already have these sourceKeys for $screen_width and $screen_height properties

**Review by utsabc — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Pull request merged**